python/pyspark/sql/connect/client.py

```diff
@@ -436,3 +568,39 @@ def _execute_and_fetch(self, req: pb2.ExecutePlanRequest) -> "pandas.DataFrame":
         if m is not None:
             df.attrs["metrics"] = self._build_metrics(m)
         return df
+
+    def _handle_error(self, rpc_error: grpc.RpcError) -> NoReturn:
```

Comment:
is it possible to match the same logic in https://github.com/apache/spark/blob/764edaf8b2e1c42a32e7bfa058cf8ee26ce02a9e/python/pyspark/sql/utils.py#L158-L197 ?

Reply:
I simplified the logic because it's going to be messy to replicate all of the exception types exactly from the SQL side. Right now the printed output looks like this for example:

```
SparkConnectException: (org.apache.spark.SparkNumberFormatException) [CAST_INVALID_INPUT] The value 'id' of the type "STRING" cannot be cast to "DOUBLE" because it is malformed. Correct the value as per the syntax, or change its target type. Use `try_cast` to tolerate malformed input and return NULL instead. If necessary set "spark.sql.ansi.enabled" to "false" to bypass this error.
== SQL(line 1, position 8) ==
select cast('id' as double) from range(10)
```